### ISSUE TITLE
JAWS, VoiceOver, and Narrator do not work correctly when accessing links in cards on APG Patterns and Fundamentals pages when using Chrome, Edge, or Safari

### ISSUE THREAD

**Issue by mcking65** [w3c/wai-aria-practices#70]
The links contained in h2 elements in the cards on the [Patterns](https://wai-aria-practices2.netlify.app/patterns/)
and [Fundamentals](https://wai-aria-practices2.netlify.app/fundamentals/)
pages exhibit some severely buggy behavior when using JAWS, Narrator, or VoiceOver in Chrome, Edge, or Safari. The behaviors are slightly different for each screen reader. NVDA is not impacted. And similar problems are not observable with JAWS or Narrator in Firefox.

I don't see any reason for the problems in the HTML; it looks perfect:

~~~
        <li class="tile tile-accordion">
                <h2 class="tile-name">
                  <a href="/patterns/accordion/">
                    <img src="/assets/img/accordion.svg" alt="">
                    <span>Accordion (Sections With Show/Hide Functionality)</span>
                  </a>
                </h2>
                <div class="tile-introduction">
        An accordion is a vertically stacked set of interactive headings that each contain a title, content snippet, or thumbnail representing a section of content.</div>
              </li>
~~~

Perhaps the buggy behaviors are related to the CSS???

Note that links contained inside headings on other sites work correctly; the problems do not appear to be caused by the link being wrapped by a heading.

#### JAWS Behavior in Chrome and Edge

Open the [patterns page](https://wai-aria-practices2.netlify.app/patterns/). With the virtual cursor active, use down arrow or 'h' to navigate to the heading for a card and press enter to activate the link contained in the heading.

Expected behavior: The target page is opened.

Actual behavior: JAWS repeats the name of the link and nothing else happens; the link is not activated. Users may think the link is broken.

Note: there are 3 ways to get around the problem: 1) press enter a second time. 2) use Tab to navigate to the link instead of using virtual cursor reading keys. 3) use the JAWS list of links to choose and activate the link.

#### Narrator behavior in Edge and Chrome

Open the [patterns page](https://wai-aria-practices2.netlify.app/patterns/). With Narrator scan mode on (capslock+s),
use down arrow or 'h' to navigate to the heading for a card and press enter to activate the link contained in the heading.

Expected behavior: The target page is opened.

Actual behaviors:
* When navigating with "h", Narrator announces "command not available" before reading the heading.
* After pressing enter, a portion of the title tag is announced, and focus seems to be set to the body; pressing insert+tab after pressing enter indicates the focus is on the body.
#### VoiceOver behavior in Chrome and Safari

Open the [patterns page](https://wai-aria-practices2.netlify.app/patterns/).
Reading with ctrl-opt-right, navigate to the second H2. Continue using ctrl-opt-right to navigate by element and navigate through several cards.

Expected behavior: The content of an h2 element is announced when the VO reading cursor is on an h2 element.

Actual behavior: Instead of reading the heading content, VO says "dimmed", e.g., VO announces "Heading 2, dimmed, 3 of 29"

**Comment by mcking65** [comment]
@smhigley, @cookiecrook, @aleventhal, is it possible you could shed some light on potential causes of this issue? Even if there are Chromium or Webkit bugs at the root of this issue, we need to find a work around before launching the new APG site on May 19.

Any insight would be deeply appreciated!!!

**Comment by scottaohara** [comment]
>Actual behavior: Instead of reading the heading content, VO says "dimmed", e.g., VO announces "Heading 2, dimmed, 3 of 29"

interestingly, if you navigate away from the content and then back to it, VO no longer says "dimmed" and instead announces the text as one would expect.

removing the CSS of the page didn't appear to help at all, but swapping the markup to be `<a href><h2>...` rather than the current `<h2><a href>...` seemed to mitigate the issue with VO/Safari/Chromium browsers.

**Comment by aleventhal** [comment]
@mcking65 I can't say quickly, but if someone files a bug at crbug.com I can have it looked at.

**Comment by mcking65** [comment]
@scottaohara  wrote:

> removing the CSS of the page didn't appear to help at all, but swapping the markup to be `<a href><h2>...` rather than the current `<h2><a href>...` seemed to mitigate the issue with VO/Safari/Chromium browsers.

Thanks Scott. Unfortunately, changing the structure like that helps VO but not JAWS or Narrator.

In this zip, [TestNestingHeadingInLink.zip](https://github.com/w3c/wai-aria-practices/files/8655821/TestNestingHeadingInLink.zip), I changed the first two cards to have the heading nested inside the link. I can verify that VoiceOver reads those first two cards as expected and announces "dimmed" for the others (until you navigate backward to them). However, the JAWS and Narrator behavior is the same for all cards.


In this zip, [NakedHTMLTestOfLinkAndHeadingNesting.zip](https://github.com/w3c/wai-aria-practices/files/8655901/NakedHTMLTestOfLinkAndHeadingNesting.zip), I stripped the HTML down to a basically empty head with only the UL in the body. The problems with JAWS and Narrator go away, regardless of whether the heading is nested in the link or vice versa.

**Comment by scottaohara** [comment]
ah that's too bad.  sorry i couldn't do a more thorough test this morning!

**Comment by mcking65** [comment]
@aleventhal I filed:
https://bugs.chromium.org/p/chromium/issues/detail?id=1323976

**Comment by richnoah** [comment]
@mcking65 this will be an item we have to work on post the launch.  There is still some discovery to complete.  Issues have been found in the HTML and how browsers are rendering it.  We had to divert resources to tackle the last-minute requests and get the website in a deploy-ready state.  I will keep this in the in-progress state so work can resume later.

**Comment by mcking65** [comment]
I have verified the workaround just landed with #93 is working!!!

Thank you @alflennik and team!!!!!
